spec/rspec/matchers/built_in/raise_error_spec.rb

```diff
@@ -1,11 +1,38 @@
 RSpec.describe "expect { ... }.to raise_error" do
   it_behaves_like("an RSpec matcher", :valid_value => lambda { raise "boom" },
                                       :invalid_value => lambda { }) do
-    let(:matcher) { raise_error }
+    let(:matcher) { raise_error Exception }
   end
 
   it "passes if anything is raised" do
-    expect {raise}.to raise_error
+    expect { raise "error" }.to raise_error "error"
+  end
+
+  it "issues a warning when used without an error class or message" do
+    expect_warning_with_call_site __FILE__, __LINE__+1, /without providing a specific error/
+    expect { raise }.to raise_error
+  end
+
+  it "can supresses the warning when configured to do so" do
+    original_value = RSpec::Expectations.configuration.warn_about_false_positives?
+    begin
+      RSpec::Expectations.configuration.warn_about_false_positives = false
+      expect { raise }.to raise_error
+    ensure
+      RSpec::Expectations.configuration.warn_about_false_positives = original_value
+    end
+  end
+
+  it 'does not issue a warning when an exception class is specified (even if it is just `Exception`)' do
+    expect { raise "error" }.to raise_error Exception
+  end
+
+  it 'does not issue a warning when a message is specified' do
+    expect { raise "error" }.to raise_error "error"
+  end
+
+  it 'does not issue a warning when a block is passed' do
+    expect { raise "error" }.to raise_error { |_| }
```

Comment:
These examples all state that no warning is issued but there are no expectations to that effect.  I understand that our rspec-support warning tracking will fail the example if there is one but I think these examples would benefit from a more explicit expectation.  I expect us to always keep the warning prevention logic but there's no guarantee it will always work the way it is -- for example, maybe we'll refactor it to do a single end-of-run check or something rather than checking in each example...and if we change it, these examples, as they currently exist, would not verify what they claim to verify.